core/model/transaction.hpp

```diff
@@ -63,30 +40,48 @@ class Transaction {
         {}
     };
 
-    T command;
     std::string hash;
     std::vector<txSignature> txSignatures;
-
+    std::string senderPubkey;
 public:
 
-    Transaction(T command):
-        command(command)
-    {}
-
-    std::string getHash() {
-        auto parser = json_parse_with_json_nlohman::JsonParse<T>();
-        return  hash::sha3_256_hex(parser.dump(command.dump()));
-    }
-
-    std::string getAsJSON() const{
-        return command.getAsJson();
+    using Object = json_parse::Object;
+    using Rule = json_parse::Rule;
+    using Type = json_parse::Type;
+
+    Transaction(
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'Transaction' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4478128063/issues/source?bid=3712491&fileBranchId=3923596#l52)